hazelcast/connection.py

```diff
@@ -487,13 +501,10 @@ def _on_auth(self, response, connection, address):
                     "Authentication status code not supported. status: %s" % status
                 )
 
-            connection.close("Failed to authenticate connection", err)
-            raise err
-        else:
-            e = response.exception()
             # This will set the exception for the pending connection future
+            connection.close("Failed to authenticate connection", err)
```

Comment:
I guess this close calls cause self._authenticate(connection) call to raise exception. Is this right?